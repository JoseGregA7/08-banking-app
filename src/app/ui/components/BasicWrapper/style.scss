.layout_main_wrapper {
    height: 100dvh;
    display: flex;
    flex-direction: column;
    width: 100%;
    align-items: center;
    background: linear-gradient(135deg, rgba(22, 22, 228, 0.189), rgba(150, 54, 9, 0.189));
}
.layout_main {
    height: 100dvh;
    display: flex;
    flex-direction: column;
    width: 100%;
    justify-content: center;
    align-items: center;
}

.layout_main__content {
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 350px;
    background-color: white;
    height: 96dvh;
    box-shadow: 1px 1px rgb(103 124 228 / 30%), 2px 2px 15px rgba(18, 48, 216, 0.6);
    border-radius: 5px;
}

.layout_main__logo {
    margin-top: 8px;
    width: 100px;
    height: 100px;
}

.layout_main__logo img {
    width: 100px;
    height: 100px;
}

.sub-text {
    font-size: 1.2rem;
    color: #666;
    padding: 0;
    margin: 0;
    margin-top: 8px;
}

.welcomeText {
    font-size: 1.8rem;
    padding: 0;
    margin: 0;
}

.layout_main__header {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    width: 100%;
    padding-bottom: 10px;
}

.back-button-wrapper {
    display: flex;
    align-items: start;
    justify-content: space-between;
    width: 100%;
    margin-top: 4px;
}

.back-button-container {
    display: flex;
    align-items: end;
    margin-left: 4px;
}

.back-button {
    font-size: 0.8rem;
    background: linear-gradient(135deg, rgba(16, 65, 226, 0.589), rgb(14, 179, 201));
    color: #fff;
    border: none;
    border-radius: 5px;
    cursor: pointer;
    transition: background-color 0.3s;

    &:hover {
        opacity: 0.8;
    }
}