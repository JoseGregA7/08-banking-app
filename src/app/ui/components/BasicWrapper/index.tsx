import { ReactNode } from 'react';
import './style.scss';
import { TermsAndConditions } from '../TermsAndConditions';
import AvatarLogout from '../AvatarComponent';

interface Props {
  children: ReactNode | ReactNode[];
  handleGoBack: () => void;
  handleEndSession: () => void;
}

const BasicWrapper = ({ children, handleGoBack, handleEndSession }: Props) => {

  return (
    <div className='layout_main_wrapper' data-testid="basic-wrapper">
      <div className='layout_main'>
        <div className='layout_main__content'>
          <header className='layout_main__header' role="banner">
            <div className='back-button-wrapper'>
              <div className='back-button-container'>
                <button
                  className='back-button'
                  onClick={handleGoBack}
                  aria-label="Volver a la página anterior"
                  data-testid="back-button"
                >
                  ↩
                </button>
              </div>
              <AvatarLogout handleEndSession={handleEndSession} />
            </div>
            <div className='layout_main__logo'>
              <img src="bank.webp" alt="Logo de GregBank, banco en línea" data-testid="bank-logo" />
            </div>
            <h1 className='welcomeText'>Bienvenido a GregBank</h1>
            <p className='sub-text'>¡Tu banco en línea, fácil y seguro!</p>
          </header>
          <main>
            {children}
          </main>
          <footer>
            <TermsAndConditions />
          </footer>
        </div>
      </div>
    </div>
  )
};

export default BasicWrapper;
