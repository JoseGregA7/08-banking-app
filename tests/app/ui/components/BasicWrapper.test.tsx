import { render, fireEvent } from '@testing-library/react';
import React from 'react';
import '@testing-library/jest-dom';
import { describe, expect, test, vitest } from 'vitest';
import BasicWrapper from '../../../../src/app/ui/components/BasicWrapper';

describe('BasicWrapper', () => {
    const mockHandleGoBack = vitest.fn();
    const mockHandleEndSession = vitest.fn();

    test('should call handleGoBack when back button is clicked', () => {
        const { getByTestId } = render(
            <BasicWrapper handleGoBack={mockHandleGoBack} handleEndSession={mockHandleEndSession}>
                <div>Test Content</div>
            </BasicWrapper>
        );
        fireEvent.click(getByTestId('back-button'));

        expect(mockHandleGoBack).toHaveBeenCalledTimes(1);
    });

    test('should call handleEndSession when AvatarLogout is interacted with', () => {
        const { getByTestId } = render(
            <BasicWrapper handleGoBack={mockHandleGoBack} handleEndSession={mockHandleEndSession}>
                <div>Test Content</div>
            </BasicWrapper>
        );

        fireEvent.click(getByTestId('avatar-logout-button'));

        expect(mockHandleEndSession).toHaveBeenCalledTimes(2);
    });

    test('should render the logo', () => {
        const { getByTestId } = render(
            <BasicWrapper handleGoBack={mockHandleGoBack} handleEndSession={mockHandleEndSession}>
                <div>Test Content</div>
            </BasicWrapper>
        );
        const logoElement = getByTestId('bank-logo');
        expect(logoElement).toBeInTheDocument();
        expect(logoElement).toHaveAttribute('alt', 'Logo de GregBank, banco en lÃ­nea');
    });

    test('should render children correctly', () => {
        const { getByText } = render(
            <BasicWrapper handleGoBack={mockHandleGoBack} handleEndSession={mockHandleEndSession}>
                <div>Test Content</div>
            </BasicWrapper>
        );

        expect(getByText('Test Content')).toBeInTheDocument();
    });

    test('should render TermsAndConditions in footer', () => {
        const { getByTestId } = render(
            <BasicWrapper handleGoBack={mockHandleGoBack} handleEndSession={mockHandleEndSession}>
                <div>Test Content</div>
            </BasicWrapper>
        );

        const termsElement = getByTestId('terms-conditions');
        expect(termsElement).toBeInTheDocument();
    });

    test('should match snapshot', () => {
        const { asFragment } = render(
            <BasicWrapper handleGoBack={mockHandleGoBack} handleEndSession={mockHandleEndSession}>
                <div>Test Content</div>
            </BasicWrapper>
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
