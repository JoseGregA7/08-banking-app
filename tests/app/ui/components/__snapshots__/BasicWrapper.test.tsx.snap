// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BasicWrapper > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="layout_main_wrapper"
    data-testid="basic-wrapper"
  >
    <div
      class="layout_main"
    >
      <div
        class="layout_main__content"
      >
        <header
          class="layout_main__header"
          role="banner"
        >
          <div
            class="back-button-wrapper"
          >
            <div
              class="back-button-container"
            >
              <button
                aria-label="Volver a la página anterior"
                class="back-button"
                data-testid="back-button"
              >
                ↩
              </button>
            </div>
            <div
              aria-label="Cerrar sesión"
              class="avatar-logout"
              data-testid="avatar-logout-wrapper"
              role="button"
            >
              <img
                alt="Avatar de usuario - Cerrar sesión"
                class="avatar"
                data-testid="avatar-logout-button"
                role="presentation"
                src="avatar.webp"
              />
            </div>
          </div>
          <div
            class="layout_main__logo"
          >
            <img
              alt="Logo de GregBank, banco en línea"
              data-testid="bank-logo"
              src="bank.webp"
            />
          </div>
          <h1
            class="welcomeText"
          >
            Bienvenido a GregBank
          </h1>
          <p
            class="sub-text"
          >
            ¡Tu banco en línea, fácil y seguro!
          </p>
        </header>
        <main>
          <div>
            Test Content
          </div>
        </main>
        <footer>
          <div>
            <a
              class="home__terms-conditions"
              data-testid="terms-conditions"
              href="#"
            >
              Términos y Condiciones
            </a>
          </div>
        </footer>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
